language/move-prover/lab/src/plot.rs

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+// Functions for reading benchmark data and converting them to graphics.
+// those results and plotting them.
+
+use crate::{
+    benchmark,
+    benchmark::{Benchmark, BenchmarkData},
+};
+use anyhow::Context;
+use clap::{App, Arg};
+use itertools::Itertools;
+use plotters::{
+    coord::types::RangedCoordu32,
+    prelude::{
+        Cartesian2d, IntoDrawingArea, IntoFont, RGBColor, Rectangle, SVGBackend, ShapeStyle, Text,
+        BLACK, BLUE, CYAN, GREEN, MAGENTA, RED, WHITE, YELLOW,
+    },
+};
+use std::collections::BTreeMap;
+
+// =====================================================================================
+// Command line interface
+
+pub fn plot_svg(args: &[String]) -> anyhow::Result<()> {
+    let is_number = |s: String| {
+        s.parse::<usize>()
+            .map(|_| ())
+            .map_err(|_| "expected number".to_string())
+    };
+    let cmd_line_parser = App::new("plot")
```

Comment:
Here, `structopt` would make it easy for `plot_svg` to take a clean configuration structure (instead of a vector of unparsed arguments). Thanks to the "flatten" option of structopt, the configuration struct can even be equipped with its own reusable command-line parsing logic:
https://github.com/facebookincubator/smt2utils/blob/master/z3tracer/src/main.rs#L18
https://github.com/facebookincubator/smt2utils/blob/master/z3tracer/src/model.rs#L23
(^^ there is only one caveat due to a minor bug in cargo doc)

Reply:
Yeah, I'm still using clap, as structopt wasn't ready when I started doing this. Should switch this and other code to it in the future.